lsp/pyright.lua

```diff
@@ -0,0 +1,71 @@
+local util = require 'lspconfig.util'
+
+local function organize_imports()
+  local params = {
+    command = 'pyright.organizeimports',
+    arguments = { vim.uri_from_bufnr(0) },
+  }
+
+  local clients = util.get_lsp_clients {
+    bufnr = vim.api.nvim_get_current_buf(),
+    name = 'pyright',
+  }
+  for _, client in ipairs(clients) do
+    client.request('workspace/executeCommand', params, nil, 0)
+  end
+end
+
+local function set_python_path(path)
+  local clients = util.get_lsp_clients {
+    bufnr = vim.api.nvim_get_current_buf(),
+    name = 'pyright',
+  }
+  for _, client in ipairs(clients) do
+    if client.settings then
+      client.settings.python = vim.tbl_deep_extend('force', client.settings.python, { pythonPath = path })
+    else
+      client.config.settings = vim.tbl_deep_extend('force', client.config.settings, { python = { pythonPath = path } })
+    end
+    client.notify('workspace/didChangeConfiguration', { settings = nil })
+  end
+end
+
+---@brief
+---
+---https://github.com/microsoft/pyright
+--
+-- `pyright`, a static type checker and language server for python
+return {
+  default_config = {
+    cmd = { 'pyright-langserver', '--stdio' },
+    filetypes = { 'python' },
+    root_markers = {
+      'pyproject.toml',
+      'setup.py',
+      'setup.cfg',
+      'requirements.txt',
+      'Pipfile',
+      'pyrightconfig.json',
+      '.git',
+    },
+    settings = {
+      python = {
+        analysis = {
+          autoSearchPaths = true,
+          useLibraryCodeForTypes = true,
+          diagnosticMode = 'openFilesOnly',
+        },
+      },
+    },
+    on_attach = function()
+      vim.api.nvim_buf_create_user_command(0, 'PyrightOrganizeImports', organize_imports, {
+        desc = 'Organize Imports',
+      })
+      vim.api.nvim_buf_create_user_command(0, 'PyrightSetPythonPath', set_python_path, {
+        desc = 'Reconfigure pyright with the provided python path',
+        nargs = 1,
+        complete = 'file',
+      })
+    end,
+  },
```

Comment:
Hey! Thanks for all your work! I checked your branch and used it today on different projects, and noticed that only pyright wasn't working because of the different keys aren't at the top level:

```suggestion
  cmd = { 'pyright-langserver', '--stdio' },
  filetypes = { 'python' },
  root_markers = {
    'pyproject.toml',
    'setup.py',
    'setup.cfg',
    'requirements.txt',
    'Pipfile',
    'pyrightconfig.json',
    '.git',
  },
  settings = {
    python = {
      analysis = {
        autoSearchPaths = true,
        useLibraryCodeForTypes = true,
        diagnosticMode = 'openFilesOnly',
      },
    },
  },
  on_attach = function()
    vim.api.nvim_buf_create_user_command(0, 'PyrightOrganizeImports', organize_imports, {
      desc = 'Organize Imports',
    })
    vim.api.nvim_buf_create_user_command(0, 'PyrightSetPythonPath', set_python_path, {
      desc = 'Reconfigure pyright with the provided python path',
      nargs = 1,
      complete = 'file',
    })
  end,
```

Looked for `default_config` in the `lps/` directory and didn't see any others.

Reply:
tysm!